src/repositories_templates.tf

```diff
@@ -42,6 +42,19 @@ resource "github_repository" "svc_template_rust" {
   name        = "svc-template-rust"
   description = "Arrow Service repository template for Rust services"
   visibility  = "public"
+  default_branch_protection = {
+    required_status_checks = {
+      strict = true
+    },
+    required_pull_request_reviews = {
+      dismiss_stale_reviews           = true
+      restrict_dismissals             = true
+      require_code_owner_reviews      = true
+      required_approving_review_count = 2
```

Comment:
Can 2 be too many? If there are small but blocking fixes, wouldn't it slow down our dev momentum?

Not suggesting a change, just something to note.